### DISCUSSION TITLE
Multiple git commits per release?

### DISCUSSION THREAD

**Discussion by rconnamacher** [intuit/auto#1949] [Q&A] [answered]
Auto's currently making two commits to git for every one release; one to edit the changelog, and one to set the git tag. Is there a way to combine them into one commit?

(I'm using the `npm` and `conventional-commits` plugins.)

```
commit  <hash> (tag: v0.2.1)
Author: <email>
Date:   <date>

    Bump version to: 0.2.1 [skip ci]

commit  <hash>
Author: <email>
Date:   <date>

    Update CHANGELOG.md [skip ci]
```

**Comment by hipstersmoothie** [comment, marked as answer]
We could probably have a small plugin that does a squash using the afterVersion hook. If you want that functionality it would be be too hard and would be a great first pr to the repo 😄 I might have time to do this for awhile but would happily answer any implementation questions

**Reply by jBouyoud** [reply]
Hi,

I could try to implement this plugin. 

Just a confirmation before, if I squash all commits on `afterVersion` hook, all tags put on the original commits will be lost. 
So the plugin must fetch and recreate all tags into the newly created commit, right ?

**Reply by hipstersmoothie** [reply]
Yeah I could see how that might get tricky! It would have to do something along those lines

**Reply by jBouyoud** [reply]
Here an implementation of this : https://github.com/intuit/auto/pull/2180

**Comment by brunonardini** [comment]
I created a [this PR](https://github.com/intuit/auto/pull/2258) solving this problem with a alternative solution. 

I already do the separate processes in a script, and I made a proposal that makes a small change, but it will allow me to generate a single commit with all the changes. I give all the details there in PR.
